core/client/fs/src/main/java/alluxio/client/block/stream/GrpcDataReader.java

```diff
@@ -141,6 +141,14 @@ public DataBuffer readChunk() throws IOException {
       return null;
     }
     mPosToRead += buffer.readableBytes();
+    if (!mStream.isClosedFromRemote()) {
+      try {
+        mStream.send(ReadRequest.newBuilder().setOffsetReceived(mPosToRead).build());
```

Comment:
It feels like the check should be part of the send method?

Reply:
Updated.